src/plugins/ruby/README.md

```diff
@@ -0,0 +1,141 @@
+    # semantic as returning -1
+  end
+
+  # the close method is currently not supported, since this will crash the 
+  # Ruby-VM if this method should be called during the VM.finializaion !!!
+  #def close(errorKey)
+  #
+  #end
+
+
+  # 'check_conf' method, called before this plugin is used for mounting to check
+  # if the supplied config is valid. Missing or invalid config settings can be
+  # 'corrected' or added.
+  # Parameter:
+  #  - errorKey: Kdb::Key, to add error information
+  #  - config: Kdb::KeySet, holds all plugin configuration setting
+  # returns:
+  #  - nil or 1 : success and the plugin configuration was NOT changed
+  #  -        0 : the plugin configuration was changed and is now OK
+  #  -       -1 : the configuration is NOT OK
+  def check_conf(errorKey, config)
+
+  end
+
+
+  # 'get' method, is called during a get cycle, to query all keys
+  #
+  # Parameter:
+  #  - returned : Kdb::KeySet, already holding keys to be manipulated or to be
+  #               filled by a storage plugin. All added keys have to have the same
+  #               key name prefix as given by parentKey.
+  #  - parentKey: Kdb::Key, defining the current Elektra path. The value of this key
+  #               holds the configuration file name to read from
+  # Returns:
+  #  - nil or 1 : on success
+  #  -        0 : OK but nothing was to do
+  #  -       -1 : failure
+  def get(returned, parentKey)
+
+  end
+
+
+  # 'set' method, is called when writing a configuration file
+  #
+  # Parameter:
+  #  - returned : Kdb::KeySet, holding all keys which should be written to the
+  #               config file.
+  #  - parentKey: Kdb::Key, defining the current Elektra path. The value of this key
+  #               holds the configuration file name to write to
+  # Returns:
+  #  - nil or 1 : on success
+  #           0 : on success with no changed keys in database
+  #          -1 : failure
+  def set(returned, parentKey)
+
+  end
+
+
+  # 'error' method, is called when some plugin had a problem during a write cycle
+  #
+  # Parameter:
+  #  - returned : Kdb::KeySet, holding all keys which should be written to the
+  #               config file.
+  #  - parentKey: Kdb::Key, defining the current Elektra path. The value of this key
+  #               holds the configuration file name to write to
+  # Returns:
+  #  - nil or 1 : on success
+  #           0 : on success with no action
+  #          -1 : failure
+  def set(returned, parentKey)
+
+  end
+
+end
+```
+
+## Known Issues
+
+- The plugin might cause a SIGSEGV crash on application shutdown. This is a result of missing
+  RubyVM de initialization, which is silently ignored as we can't determine if the Ruby VM
+  is still needed or not (libelektra might be used by an Ruby application).
+- The plugin does not work correctly with Ruby version managers like `rbenv` or `rvm` and is
+  intended to be used with the Ruby ecosystem installed in system context.
+- Automated test cases are missing
+
```

Comment:
Maybe you could also instructions on how to manually test the plugin. As far as I can tell you already added some tests in the folder `ruby/spec`. I think it would make sense to tell a user how she or he can invoke them.

Reply:
These tests are only meant for testing the shell_vars plugin and helped me during development (TDD). The actual plugin is not used by the tests.